manager.py

```diff
@@ -0,0 +1,160 @@
+import docker
+import psycopg2
+import json
+import time
+
+
+def get_env_vars(client):
+    global postgres_user
+    global postgres_pass
+    global postgres_db
+
+    container = client.containers.get("citus_master")
+
+    env_vars = container.exec_run(cmd='env')
+
+    for each_line in env_vars.splitlines():
+        if 'POSTGRES' in each_line:
+            (key, val) = each_line.split("=")
+            if key == "POSTGRES_USER":
+                postgres_user = val
+            elif key == "POSTGRES_PASSWORD":
+                postgres_pass = val
+            elif key == "POSTGRES_DB":
+                postgres_db = val
+
+
+def connect_to_db(client):
+    conn = None
+    a = 0
+
+    get_env_vars(client)
+
+    while a < 10 and conn is None:
+        time.sleep(2)
+        try:
+            conn = psycopg2.connect("dbname=%s user=%s host=%s password=%s" %
+                                    (postgres_db, postgres_user, "citus_master", postgres_pass))
+        except:
+            print("I am unable to connect")
+            a = a + 1
+
+    return conn
+
+
+def add_workers(conn, workers):
+    for worker in workers:
+        add_worker(conn, worker, 5432)
+    conn.commit()
+
+
+def add_worker(conn, host, port):
+    cur = conn.cursor()
+    worker_dict = ({"host": host, "port": port})
+    try:
+        cur.execute("""SELECT master_add_node(%(host)s, %(port)s)""", worker_dict)
+    except:
+        print("I can't add worker to the coordinator!")
+
+
+def remove_worker(conn, host, port):
+    cur = conn.cursor()
+    worker_dict = ({"host": host, "port": port})
+    try:
+        cur.execute("""SELECT master_remove_node(%(host)s, %(port)s)""", worker_dict)
+    except:
+        print("I can't remove worker from the coordinator!")
+
+
+def get_workers(conn):
+    cur = conn.cursor()
+    workers = None
+
+    try:
+        cur.execute("""SELECT master_get_active_worker_nodes()""")
+        workers = cur.fetchall()
+    except:
+        print("I can't fetch active workers!")
+
+    return workers
+
+
+def check_cluster(client):
+    containers = client.containers.list()
+    conn = None
+    worker_set = set()
+    workers = set()
+    client_API = docker.APIClient(base_url=end_point)
+
+    for container in containers:
+        c_name = container.name
+
+        status = client_API.inspect_container(c_name)
+
+        if 'Health' in status['State']:
+            health_status = status['State']['Health']['Status']
+            if health_status == 'healthy':
+                if c_name == 'citus_master':
```

Comment:
We _cannot_ use the name of the worker as an identifier here. That's what labels are for. The workers are only named `citus_worker` if the `-p` flag is specified as `citus`. They could easily be `foo_worker`.

Reply:
Which is to say, figure out how to get the labels of an instance and look for the citus role worker label. If it's not present, don't do anything. If it is, call `master_add_node`.